cmd/juju/action/show.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package action
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/utils/featureflag"
+	"gopkg.in/juju/names.v3"
+	"gopkg.in/yaml.v2"
+
+	"github.com/juju/juju/apiserver/params"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/feature"
+)
+
+type showCommand struct {
+	ActionCommandBase
+
+	applicationTag names.ApplicationTag
+	actionName     string
+
+	out cmd.Output
+}
+
+var showActionDoc = `
+Show detailed information about an action on the target application.
+
+Examples:
+    juju show-action postgresql backup
+
+See also:
+    list-actions
+    run-action
+`
+
+// NewShowCommand returns a command to print action information.
+func NewShowCommand() cmd.Command {
+	return modelcmd.Wrap(&showCommand{})
+}
+
+func (c *showCommand) Init(args []string) error {
+	switch len(args) {
+	case 0:
+		return errors.New("no application name specified")
+	case 1:
+		return errors.New("no action name specified")
+	case 2:
+		appName := args[0]
+		if !names.IsValidApplication(appName) {
+			return errors.Errorf("invalid application name %q", appName)
+		}
+		c.applicationTag = names.NewApplicationTag(appName)
+		c.actionName = args[1]
+		return nil
+	default:
+		return cmd.CheckEmpty(args[2:])
+	}
+}
+
+func (c *showCommand) Info() *cmd.Info {
+	info := jujucmd.Info(&cmd.Info{
+		Name:    "show-action",
+		Args:    "<application name> <action name>",
+		Purpose: "Shows detailed information about an action.",
+		Doc:     showActionDoc,
+	})
+	if featureflag.Enabled(feature.JujuV3) {
+		info.Doc = strings.Replace(info.Doc, "run-action", "run", -1)
+	}
+	return info
+}
+
+func (c *showCommand) Run(ctx *cmd.Context) error {
+	api, err := c.NewActionAPIClient()
```

Comment:
It's a lot easier to test if the api client is retrieved from a command method, traditionally something along the lines of `newClientF`. This way you could also specify in interface what api methods you command needs :)

Reply:
Yeah, this method is on the base command. Didn't want to change all the existing commands in this PR.